api/turing/middleware/openapi_validation.go

```diff
@@ -30,34 +32,36 @@ func NewOpenAPIValidation(
 	openapiYamlFile string,
 	options OpenAPIValidationOptions,
 ) (*OpenAPIValidation, error) {
-	loader := &openapi3.SwaggerLoader{
+	// Get Swagger specs
+	loader := &openapi3.Loader{
 		IsExternalRefsAllowed: true,
 	}
-	swagger, err := loader.LoadSwaggerFromFile(openapiYamlFile)
+	swagger, err := loader.LoadFromFile(openapiYamlFile)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Error loading swagger spec\n: %s", err)
 	}
 
+	// Handle options
+	openAPIFilterOpts := &openapi3filter.Options{}
+	if options.IgnoreAuthentication {
+		openAPIFilterOpts.AuthenticationFunc = openapi3filter.NoopAuthenticationFunc
+	}
 	if options.IgnoreServers {
 		swagger.Servers = nil
 	}
-	router := openapi3filter.NewRouter().WithSwagger(swagger)
-
-	opts := &openapi3filter.Options{}
 
-	if options.IgnoreAuthentication {
-		// when IgnoreAuthentication is true, the authentication function always succeed i.e returns nil error.
-		opts.AuthenticationFunc = func(_ context.Context, _ *openapi3filter.AuthenticationInput) error {
-			return nil
-		}
+	// Create router
+	var router routers.Router
+	if router, err = legacyrouter.NewRouter(swagger); err != nil {
```

Comment:
Why do we still need to use `legacyrouter`?

Reply:
No strong reason. The alternative is `gorilla/mux` router, but in comparison, the legacy router supposedly captures more granular errors: https://pkg.go.dev/github.com/getkin/kin-openapi@v0.75.0/routers/legacy

On the other hand, the `gorilla/mux` router can match variables in the paths with extensions like `{id}.json`. But we don't really use it.

Since the `legacyrouter` is closer to what we were using earlier, I picked it. It's a one line change to opt for the other, if we need to later.